soul-admin/src/main/java/org/dromara/soul/admin/service/impl/DashboardUserServiceImpl.java

```diff
@@ -88,6 +88,10 @@ public DashboardUserServiceImpl(final DashboardUserMapper dashboardUserMapper, f
     @Override
     @Transactional(rollbackFor = Exception.class)
     public int createOrUpdate(final DashboardUserDTO dashboardUserDTO) {
+        DashboardUserDO existedDashboardUserDO = dashboardUserMapper.selectByUserName(dashboardUserDTO.getUserName());
+        if (Objects.nonNull(existedDashboardUserDO) && !existedDashboardUserDO.getId().equals(dashboardUserDTO.getId())) {
+            return -1;
+        }
```

Comment:
![image](https://user-images.githubusercontent.com/46160170/107151190-3a5e3800-699c-11eb-914d-1d7e462d83d5.png)

Wouldn't it be better to change it this way, so that the Controller doesn't have to change